build.assets/Dockerfile-arm

```diff
@@ -1,8 +1,75 @@
-ARG BUILDBOX_VERSION
-FROM public.ecr.aws/gravitational/teleport-buildbox:$BUILDBOX_VERSION
+# This Dockerfile is used to build Teleport on ARM only.
+# We are using the official Debian 11 image as a base image
+# because the finaly binary must be compatible with distroless
+# images that are also Debian 11 based: https://github.com/GoogleContainerTools/distroless
+FROM docker.io/library/debian:11
```

Comment:
We should make sure the GHA workflows are using `docker/login-action` to authenticate to Docker Hub so we don't run into any rate limiting issues.

Reply:
This docker file is used by Drone (at least for now), so I'm not sure if this applies.

Reply:
It will be used on github too until everything is migrated to a common centos7 buildbox (or whatever replaces that), but the workflow in which I will use this has the dockerhub login so should be fine.